payments-core/src/main/java/com/stripe/android/paymentsheet/PaymentSheetContract.kt

```diff
@@ -32,19 +33,39 @@ internal class PaymentSheetContract :
     }
 
     @Parcelize
-    internal data class Args(
-        val clientSecret: ClientSecret,
-        @ColorInt val statusBarColor: Int?,
-        val config: PaymentSheet.Configuration?
+    data class Args @VisibleForTesting internal constructor(
+        internal val clientSecret: ClientSecret,
+        internal val config: PaymentSheet.Configuration?,
+        @ColorInt val statusBarColor: Int? = null,
     ) : ActivityStarter.Args {
         val googlePayConfig: PaymentSheet.GooglePayConfiguration? get() = config?.googlePay
         val isGooglePayEnabled: Boolean
             get() = googlePayConfig != null && clientSecret is PaymentIntentClientSecret
 
-        internal companion object {
+        companion object {
             internal fun fromIntent(intent: Intent): Args? {
                 return intent.getParcelableExtra(EXTRA_ARGS)
             }
+
+            fun createPaymentIntentArgs(
+                clientSecret: String,
+                config: PaymentSheet.Configuration? = null,
+                statusBarColor: Int? = null,
```

Comment:
I don't think we shouldn't expose this as a parameter. Instead, you can get it from the `context` instance passed to `createIntent`:

```
    override fun createIntent(
        context: Context,
        input: Args
    ): Intent {
        val statusBarColor = (context as? Activity).window?.statusBarColor
        return Intent(context, PaymentSheetActivity::class.java)
            .putExtra(EXTRA_ARGS, input.copy(statusBarColor = statusBarColor))
    }
```

Reply:
THe context is not an activity in jetpack compose.

Reply:
Interesting - what is it?

Reply:
oh, I was incorrect.  I am fixing it up.